src/http/common.cr

```diff
@@ -241,17 +241,18 @@ module HTTP
     referer
     user-agent
   )
+    .map { |header| {header.to_slice, header} }
+    .to_h
```

Comment:
Array allocation can be avoided:

```suggestion
    .to_h { |header| {header.to_slice, header} }
```

Reply:
Allocation of a single object (well, technically 2 since the array contains a reference to a heap-allocated buffer) in a constant only saves about 10 nanoseconds over the life of the program, _however_ that's really slick and TIL `to_h` accepts a block! 😄